dd-sdk-android/src/main/kotlin/com/datadog/android/v2/core/internal/net/DataOkHttpUploader.kt

```diff
@@ -34,7 +34,18 @@ internal class DataOkHttpUploader(
         batch: List<ByteArray>,
         batchMeta: ByteArray?
     ): UploadStatus {
-        val request = requestFactory.create(context, batch, batchMeta)
+        val request = try {
+            requestFactory.create(context, batch, batchMeta)
+        } catch (e: Exception) {
```

Comment:
unfortunately we don't. For example in case of SR it can be because of a bad payload format but in other cases it may be something else. I followed what `iOS` is doing also and I saw they also have it generic.